### ISSUE TITLE
Frigate crashes after a few minutes in Home Assistant and repeats when manually restarted

### ISSUE THREAD

**Issue by snarlingllama** [blakeblackshear/frigate#4513]
### Describe the problem you are having

Frigate appears to be crashing after a few minutes of launching. This is an issue that started about 2 days ago. I did not notice any syntax issues in my configuration (I turned on automatic updates about 2 days ago), but I could be wrong.

### Version

0.11.1

### Frigate config file

```yaml
mqtt:
  host: <IP>
  port: 1883
  topic_prefix: frigate
  client_id: frigate
  user: mqtt
  password: <password>
  stats_interval: 30
 
detectors:
  coral1:
    type: edgetpu
    device: usb:0
  coral2:
    type: edgetpu
    device: usb:1
  
cameras:
  backyard_north:
    ffmpeg:
      inputs:
        - path: <confirmed-working-rtsp-url>
          roles:
            - detect
    detect:
      width: 1280
      height: 720
      fps: 5
      enabled: True
      stationary:
        interval: 3
        threshold: 50
    objects:
      track:
        - person
        - dog
        - cat
      filters:
        person:
          min_score: 0.5
    snapshots:
      enabled: True
      timestamp: True
      bounding_box: True
      retain:
        default: 2
        
  backyard_south:
    ffmpeg:
      inputs:
        - path: <confirmed-working-rtsp-url>
          roles:
            - detect
    detect:
      width: 1280
      height: 720
      fps: 5
      enabled: True
      stationary:
        interval: 3
        threshold: 50
    objects:
      track:
        - person
        - dog
        - cat
      filters:
        person:
          min_score: 0.5
    snapshots:
      enabled: True
      timestamp: True
      bounding_box: True
      retain:
        default: 2
        
  backyard_rear:
    ffmpeg:
      inputs:
        - path: <confirmed-working-rtsp-url>
          roles:
            - detect
    detect:
      width: 1280
      height: 720
      fps: 5
      enabled: True
      stationary:
        interval: 3
        threshold: 50
    objects:
      track:
        - bear
        - dog
        - cat
    snapshots:
      enabled: True
      timestamp: True
      bounding_box: True
      retain:
        default: 2

  pool:
    ffmpeg:
      inputs:
        - path: <confirmed-working-rtsp-url>
          roles:
            - detect
    detect:
      width: 1280
      height: 720
      fps: 5
      enabled: True
      stationary:
        interval: 3
        threshold: 50
    objects:
      track:
        - person
      filters:
        person:
          min_score: 0.8
    snapshots:
      enabled: True
      timestamp: True
      bounding_box: True
      retain:
        default: 2
        
  front_door:
    ffmpeg:
      inputs:
        - path: <confirmed-working-rtsp-url>
          roles:
            - detect
    detect:
      width: 1280
      height: 720
      fps: 5
      enabled: True
      stationary:
        interval: 3
        threshold: 50
    objects:
      track:
        - person
        - dog
      filters:
        person:
          min_score: 0.6
    snapshots:
      enabled: True
      timestamp: True
      bounding_box: True
      retain:
        default: 2
```

### Relevant log output

```shell
[s6-init] making user provided files available at /var/run/s6/etc...exited 0.
[s6-init] ensuring user provided files have correct perms...exited 0.
[fix-attrs.d] applying ownership & permissions fixes...
[fix-attrs.d] done.
[cont-init.d] executing container initialization scripts...
[cont-init.d] done.
[services.d] starting services
[services.d] done.
[2022-11-25 12:49:26] frigate.app                    INFO    : Starting Frigate (0.11.1-2eada21)
[2022-11-25 12:49:26] frigate.app                    INFO    : Creating directory: /tmp/cache
Starting migrations
[2022-11-25 12:49:26] peewee_migrate                 INFO    : Starting migrations
There is nothing to migrate
[2022-11-25 12:49:26] peewee_migrate                 INFO    : There is nothing to migrate
[2022-11-25 12:49:26] detector.coral1                INFO    : Starting detection process: 216
[2022-11-25 12:49:26] frigate.mqtt                   INFO    : Turning off snapshots for backyard_rear via mqtt
[2022-11-25 12:49:26] frigate.edgetpu                INFO    : Attempting to load TPU as usb:0
[2022-11-25 12:49:26] frigate.app                    INFO    : Output process started: 219
[2022-11-25 12:49:26] detector.coral2                INFO    : Starting detection process: 217
[2022-11-25 12:49:26] frigate.app                    INFO    : Camera processor started for backyard_north: 223
[2022-11-25 12:49:26] frigate.app                    INFO    : Camera processor started for backyard_south: 226
[2022-11-25 12:49:26] frigate.app                    INFO    : Camera processor started for backyard_rear: 230
Process detector:coral2:
[2022-11-25 12:49:26] frigate.edgetpu                INFO    : Attempting to load TPU as usb:1
[2022-11-25 12:49:26] frigate.edgetpu                ERROR   : No EdgeTPU was detected. If you do not have a Coral device yet, you must configure CPU detectors.
Traceback (most recent call last):
  File "/usr/lib/python3/dist-packages/tflite_runtime/interpreter.py", line 160, in load_delegate
    delegate = Delegate(library, options)
  File "/usr/lib/python3/dist-packages/tflite_runtime/interpreter.py", line 119, in __init__
    raise ValueError(capture.message)
ValueError
During handling of the above exception, another exception occurred:
Traceback (most recent call last):
  File "/usr/lib/python3.9/multiprocessing/process.py", line 315, in _bootstrap
    self.run()
  File "/usr/lib/python3.9/multiprocessing/process.py", line 108, in run
    self._target(*self._args, **self._kwargs)
  File "/opt/frigate/frigate/edgetpu.py", line 135, in run_detector
    object_detector = LocalObjectDetector(
  File "/opt/frigate/frigate/edgetpu.py", line 43, in __init__
    edge_tpu_delegate = load_delegate("libedgetpu.so.1.0", device_config)
  File "/usr/lib/python3/dist-packages/tflite_runtime/interpreter.py", line 162, in load_delegate
    raise ValueError('Failed to load delegate from {}\n{}'.format(
ValueError: Failed to load delegate from libedgetpu.so.1.0
[2022-11-25 12:49:26] ws4py                          INFO    : Using epoll
[2022-11-25 12:49:26] frigate.app                    INFO    : Camera processor started for pool: 234
[2022-11-25 12:49:26] frigate.mqtt                   INFO    : Turning off detection for pool via mqtt
[2022-11-25 12:49:26] frigate.app                    INFO    : Camera processor started for front_door: 236
[2022-11-25 12:49:27] frigate.app                    INFO    : Capture process started for backyard_north: 238
[2022-11-25 12:49:27] frigate.app                    INFO    : Capture process started for backyard_south: 240
[2022-11-25 12:49:27] frigate.app                    INFO    : Capture process started for backyard_rear: 246
[2022-11-25 12:49:27] frigate.app                    INFO    : Capture process started for pool: 250
[2022-11-25 12:49:27] frigate.app                    INFO    : Capture process started for front_door: 256
[2022-11-25 12:49:27] ws4py                          INFO    : Using epoll
[2022-11-25 12:49:30] frigate.edgetpu                INFO    : TPU found
[2022-11-25 12:49:47] frigate.watchdog               INFO    : Detection appears to have stopped. Exiting frigate...
[2022-11-25 12:49:47] frigate.app                    INFO    : Stopping...
[2022-11-25 12:49:47] ws4py                          INFO    : Closing all websockets with [1001] 'Server is shutting down'
[2022-11-25 12:49:47] frigate.stats                  INFO    : Exiting watchdog...
[2022-11-25 12:49:47] frigate.record                 INFO    : Exiting recording cleanup...
[2022-11-25 12:49:47] frigate.events                 INFO    : Exiting event cleanup...
[2022-11-25 12:49:47] frigate.watchdog               INFO    : Exiting watchdog...
[2022-11-25 12:49:47] frigate.object_processing      INFO    : Exiting object processor...
[2022-11-25 12:49:47] frigate.record                 INFO    : Exiting recording maintenance...
[2022-11-25 12:49:57] frigate.events                 INFO    : Exiting event processor...
[2022-11-25 12:49:57] peewee.sqliteq                 INFO    : writer received shutdown request, exiting.
[2022-11-25 12:49:57] root                           INFO    : Waiting for detection process to exit gracefully...
[2022-11-25 12:49:58] root                           INFO    : Waiting for detection process to exit gracefully...
```

### Operating system

HassOS

### Install method

HassOS Addon

### Coral version

USB

### Any other information that may be helpful

Home Assistant 2022.11.4
Supervisor 2022.11.2
Operating System 9.3
Frontend 20221108.0 - latest

Two USB Corals, but the logs seem to indicate Frigate is having difficulties finding them, which is a new issue (nothing has changed since its last working condition two days ago.

When I attempt to download a diagnostics file from the integration, I get a "failed - server problem" error in my browser download view.

**Comment by NickM-27** [comment]
Are you using the full access version of the addon with protection mode disabled?

**Comment by snarlingllama** [comment]
@NickM-27 I thought I was using the full access version, but it does not appear so (see attached screenshot).

![image](https://user-images.githubusercontent.com/23412697/204050557-4d7abaf6-c6f4-4c19-9736-30c8b8dd6ef5.png)

**Comment by NickM-27** [comment]
Recommend doing that then along with disabling protected mode after

**Comment by snarlingllama** [comment]
@NickM-27 Thank you for the response. Any best practice advice on how to do this? Because my plan would be just to remove my existing addon, then install the full access version. Would that cause any issues?

**Comment by NickM-27** [comment]
No it won't cause any issues.

**Comment by snarlingllama** [comment]
Just the addon or the integration as well? Assuming we're just focusing on the addon.

**Comment by NickM-27** [comment]
You may need to remove and readd the integration since the host name will be different

**Comment by snarlingllama** [comment]
@NickM-27 I might need to reboot (waiting on a zwave heal to complete), but Frigate is still crashing. I confirmed I now have the full access version installed with protected mode disabled. I also uninstalled/reinstalled the integration and provided it with the new Frigate host URL.

Here are the logs:

s.d] done.
[cont-init.d] executing container initialization scripts...
[cont-init.d] done.
[services.d] starting services
[services.d] done.
[2022-11-25 15:21:10] frigate.app                    INFO    : Starting Frigate (0.11.1-2eada21)
[2022-11-25 15:21:10] frigate.app                    INFO    : Creating directory: /tmp/cache
Starting migrations
[2022-11-25 15:21:10] peewee_migrate                 INFO    : Starting migrations
There is nothing to migrate
[2022-11-25 15:21:10] peewee_migrate                 INFO    : There is nothing to migrate
[2022-11-25 15:21:10] frigate.mqtt                   INFO    : Turning off snapshots for backyard_rear via mqtt
[2022-11-25 15:21:10] detector.coral1                INFO    : Starting detection process: 216
[2022-11-25 15:21:10] frigate.edgetpu                INFO    : Attempting to load TPU as usb:0
[2022-11-25 15:21:10] detector.coral2                INFO    : Starting detection process: 217
[2022-11-25 15:21:10] frigate.app                    INFO    : Output process started: 219
[2022-11-25 15:21:10] ws4py                          INFO    : Using epoll
[2022-11-25 15:21:11] frigate.app                    INFO    : Camera processor started for backyard_north: 225
[2022-11-25 15:21:11] frigate.mqtt                   INFO    : Turning off detection for pool via mqtt
[2022-11-25 15:21:11] frigate.app                    INFO    : Camera processor started for backyard_south: 228
[2022-11-25 15:21:11] frigate.app                    INFO    : Camera processor started for backyard_rear: 232
[2022-11-25 15:21:10] frigate.edgetpu                INFO    : Attempting to load TPU as usb:1
Process detector:coral2:
[2022-11-25 15:21:11] frigate.edgetpu                ERROR   : No EdgeTPU was detected. If you do not have a Coral device yet, you must configure CPU detectors.
Traceback (most recent call last):
  File "/usr/lib/python3/dist-packages/tflite_runtime/interpreter.py", line 160, in load_delegate
    delegate = Delegate(library, options)
  File "/usr/lib/python3/dist-packages/tflite_runtime/interpreter.py", line 119, in __init__
    raise ValueError(capture.message)
ValueError
During handling of the above exception, another exception occurred:
Traceback (most recent call last):
  File "/usr/lib/python3.9/multiprocessing/process.py", line 315, in _bootstrap
    self.run()
  File "/usr/lib/python3.9/multiprocessing/process.py", line 108, in run
    self._target(*self._args, **self._kwargs)
  File "/opt/frigate/frigate/edgetpu.py", line 135, in run_detector
    object_detector = LocalObjectDetector(
  File "/opt/frigate/frigate/edgetpu.py", line 43, in __init__
    edge_tpu_delegate = load_delegate("libedgetpu.so.1.0", device_config)
  File "/usr/lib/python3/dist-packages/tflite_runtime/interpreter.py", line 162, in load_delegate
    raise ValueError('Failed to load delegate from {}\n{}'.format(
ValueError: Failed to load delegate from libedgetpu.so.1.0
[2022-11-25 15:21:11] frigate.app                    INFO    : Camera processor started for pool: 235
[2022-11-25 15:21:11] frigate.app                    INFO    : Camera processor started for front_door: 239
[2022-11-25 15:21:11] frigate.app                    INFO    : Capture process started for backyard_north: 240
[2022-11-25 15:21:11] frigate.app                    INFO    : Capture process started for backyard_south: 242
[2022-11-25 15:21:11] frigate.app                    INFO    : Capture process started for backyard_rear: 247
[2022-11-25 15:21:11] frigate.app                    INFO    : Capture process started for pool: 250
[2022-11-25 15:21:11] frigate.app                    INFO    : Capture process started for front_door: 259
[2022-11-25 15:21:11] ws4py                          INFO    : Using epoll
[2022-11-25 15:21:14] frigate.edgetpu                INFO    : TPU found
[2022-11-25 15:21:20] ws4py                          INFO    : Managing websocket [Local => 127.0.0.1:5002 | Remote => 127.0.0.1:34176]
[2022-11-25 15:21:27] ws4py                          INFO    : Terminating websocket [Local => 127.0.0.1:5002 | Remote => 127.0.0.1:34176]
[2022-11-25 15:21:31] frigate.watchdog               INFO    : Detection appears to have stopped. Exiting frigate...
[2022-11-25 15:21:31] frigate.app                    INFO    : Stopping...
[2022-11-25 15:21:31] ws4py                          INFO    : Closing all websockets with [1001] 'Server is shutting down'
[2022-11-25 15:21:31] frigate.watchdog               INFO    : Exiting watchdog...
[2022-11-25 15:21:31] frigate.stats                  INFO    : Exiting watchdog...
[2022-11-25 15:21:31] frigate.events                 INFO    : Exiting event cleanup...
[2022-11-25 15:21:31] frigate.record                 INFO    : Exiting recording cleanup...
[2022-11-25 15:21:31] frigate.object_processing      INFO    : Exiting object processor...
[2022-11-25 15:21:31] frigate.record                 INFO    : Exiting recording maintenance...
[2022-11-25 15:21:41] frigate.events                 INFO    : Exiting event processor...
[2022-11-25 15:21:41] peewee.sqliteq                 INFO    : writer received shutdown request, exiting.
[2022-11-25 15:21:41] root                           INFO    : Waiting for detection process to exit gracefully...
[2022-11-25 15:21:43] root                           INFO    : Waiting for detection process to exit gracefully...

**Comment by NickM-27** [comment]
The zwave dongle maybe using too much power and the coral doesn't have enough to run

**Comment by snarlingllama** [comment]
@NickM-27 Very curious. I have been wondering about this as I am going through a rather hefty troubleshooting task with zwave.

My USB set up is:

Direct to RPi4:
1. SSD
2. ZWave Stick

Direct to USB powered hub:
1. Conbee II
2. Google Coral 1
3. Google Coral 2

Anyway via the system logs to identify an under-powering issue?

**Comment by NickM-27** [comment]
The corals should identify themselves as Google Inc. otherwise they aren't initialized.

**Comment by snarlingllama** [comment]
Does the initialization happen automatically when the addon is launched, or do I need to execute an action?

**Comment by NickM-27** [comment]
It will happen when frigate starts yes

**Comment by snarlingllama** [comment]
Alright I still have no rebooted, but went ahead and restarted the addon. Not seeing the errors this time, but once in a while it will detect a human or two before shutting down.

[s6-init] making user provided files available at /var/run/s6/etc...exited 0.
[s6-init] ensuring user provided files have correct perms...exited 0.
[fix-attrs.d] applying ownership & permissions fixes...
[fix-attrs.d] done.
[cont-init.d] executing container initialization scripts...
[cont-init.d] done.
[services.d] starting services
[services.d] done.
[2022-11-25 17:00:00] frigate.app                    INFO    : Starting Frigate (0.11.1-2eada21)
[2022-11-25 17:00:01] frigate.app                    INFO    : Creating directory: /tmp/cache
Starting migrations
[2022-11-25 17:00:01] peewee_migrate                 INFO    : Starting migrations
There is nothing to migrate
[2022-11-25 17:00:01] peewee_migrate                 INFO    : There is nothing to migrate
[2022-11-25 17:00:01] frigate.app                    INFO    : Output process started: 220
[2022-11-25 17:00:01] frigate.app                    INFO    : Camera processor started for backyard_north: 222
[2022-11-25 17:00:01] detector.coral2                INFO    : Starting detection process: 217
[2022-11-25 17:00:01] frigate.app                    INFO    : Camera processor started for backyard_south: 224
[2022-11-25 17:00:01] ws4py                          INFO    : Using epoll
[2022-11-25 17:00:01] frigate.edgetpu                INFO    : Attempting to load TPU as usb:1
[2022-11-25 17:00:01] frigate.app                    INFO    : Camera processor started for backyard_rear: 228
[2022-11-25 17:00:01] frigate.app                    INFO    : Camera processor started for pool: 230
[2022-11-25 17:00:01] frigate.app                    INFO    : Camera processor started for front_door: 234
[2022-11-25 17:00:01] frigate.app                    INFO    : Capture process started for backyard_north: 237
[2022-11-25 17:00:01] frigate.app                    INFO    : Capture process started for backyard_south: 238
[2022-11-25 17:00:01] frigate.app                    INFO    : Capture process started for backyard_rear: 240
[2022-11-25 17:00:01] frigate.app                    INFO    : Capture process started for pool: 245
[2022-11-25 17:00:01] frigate.app                    INFO    : Capture process started for front_door: 248
[2022-11-25 17:00:01] frigate.mqtt                   INFO    : Turning off snapshots for backyard_rear via mqtt
[2022-11-25 17:00:01] frigate.mqtt                   INFO    : Turning off detection for pool via mqtt
[2022-11-25 17:00:02] ws4py                          INFO    : Using epoll
[2022-11-25 17:00:01] detector.coral1                INFO    : Starting detection process: 216
[2022-11-25 17:00:01] frigate.edgetpu                INFO    : Attempting to load TPU as usb:0
[2022-11-25 17:00:07] frigate.edgetpu                INFO    : TPU found
[2022-11-25 17:00:08] frigate.edgetpu                INFO    : TPU found
[2022-11-25 17:00:51] ws4py                          INFO    : Managing websocket [Local => 127.0.0.1:5002 | Remote => 127.0.0.1:43992]
[2022-11-25 17:01:00] ws4py                          INFO    : Terminating websocket [Local => 127.0.0.1:5002 | Remote => 

Not seeing anything to indicate recognition as Google Inc though.

**Comment by NickM-27** [comment]
It'll be in your devices list on HA OS. This definitely sounds like a power issue though.

**Comment by snarlingllama** [comment]
I can pick up a new USB to see if that is the culprit. What about a resource issue? My CPU is at 62% right now, but its usually around 80% - 89% when Frigate is running. Perhaps its crashing when the CPU gets high.

Strangely, I do not see any TPU or Google Inc devices under Devices. I searched without any filters and filtering on Frigate (which only shows the servers and my 4 cameras).

I'll post back in a bit when I run a full shutdown and restart. If that doesn't work, and there's no additional logs, I'll try a new USB hub.

Really appreciate your feedback!

**Comment by NickM-27** [comment]
I think you're looking in the wrong place, I'm talking about hardware section not the devices section in HA core settings. 

And I suppose it could be CPU related but not 100% sure

**Comment by snarlingllama** [comment]
I was looking in the wrong place. Still not seeing anything under System > Hardware > All Hardware, but I will expand each section again to be sure.  I too am beginning to think is a power issue on account of my zwave issues as well. 

Next steps for me, barring any new, interesting logs, is:

1. Try a new USB powered hub (FYSA, I currently have a Sabrent 4-port USB 3.0 hub)
2. Finally get the system fully rebooted

**Comment by snarlingllama** [comment]
I was able to reboot the server and Frigate loaded fine without issue. It has been running for about 30mins now without issue. I ordered a new USB 3.1 powered hub in case my hub is faulty.
